storage/range_tree.go

```diff
@@ -25,37 +25,27 @@ import (
 )
 
 type (
-	// RangeTree represents holds the metadata about the RangeTree.  There is
-	// only one for the whole cluster.
-	rangeTree proto.RangeTree
-	// RangeTreeNode represents a node in the RangeTree, each range
-	// has only a single node.
-	rangeTreeNode proto.RangeTreeNode
+	// cachedNode is an in memory cache for use during range tree manipulations.
+	cachedNode struct {
+		node    *proto.RangeTreeNode
+		updated bool
+	}
+	// RangeTree is used to hold the relevant context information for any
+	// operations on the range tree.
+	treeContext struct {
+		db          *client.KV
+		tree        *proto.RangeTree
+		treeUpdated bool
```

Comment:
Would probably make this just "updated" or "dirty"; no need to prefix with "tree" here.